app/models/manageiq/providers/cloud_manager/provision_workflow.rb

```diff
@@ -26,9 +26,18 @@ def allowed_cloud_subnets(_options = {})
   end
 
   def allowed_cloud_networks(_options = {})
+    return {} unless (src_obj = provider_or_tenant_object)
+
     source = load_ar_obj(get_source_vm)
     targets = get_targets_for_source(source, :cloud_filter, CloudNetwork, 'cloud_network_id')
-    allowed_ci(:cloud_network, [:availability_zone], targets.map(&:id))
+
+    if src_obj.all_cloud_networks.nil?
+      return allowed_ci(:cloud_network, [:availability_zone], targets.map(&:id))
+    else
+      src_obj.all_cloud_networks.each_with_object({}) do |cn, hash|
```

Comment:
This fixes the tests but is it really addressing the original issue of rbac filtering on this element?